### PR TITLE
nextcloudPackages: added e2e app

### PR DESCRIPTION
**Pull request by flandweber**
## Description of changes

added e2e nextcloud app

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ping @Ma27 as maintainer

### CODE REVIEW

File: pkgs/servers/nextcloud/packages/26.json [hunk 1]
```diff
@@ -49,6 +49,16 @@
       "agpl"
     ]
   },
+  "end_to_end_encryption": {
+    "sha256": "19vp0ggllplm84hwaqwn95122hsgfglmlk2lbwlgsjv8d36fv1wr",
+    "url": "https://github.com/nextcloud-releases/end_to_end_encryption/releases/download/v1.12.5/end_to_end_encryption-v1.12.5.tar.gz",
+    "version": "1.12.5",
```
Comment (FliegendeWurst) [inline pkgs/servers/nextcloud/packages/26.json:55]: Why not version 1.13.1? https://github.com/nextcloud/end_to_end_encryption/releases/tag/v1.13.1

Comment (BootRhetoric) [reply pkgs/servers/nextcloud/packages/26.json:55]: because nextcloud 26 is still on 12.5: https://apps.nextcloud.com/apps/end_to_end_encryption

The generate.sh script added those lines automatically (See [README.md](https://github.com/NixOS/nixpkgs/tree/master/pkgs/servers/nextcloud/packages))

Comment (FliegendeWurst) [reply pkgs/servers/nextcloud/packages/26.json:55]: Ah, okay.

File: pkgs/servers/nextcloud/packages/generate.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env nix-shell
-#! nix-shell -I nixpkgs=../../../.. -i bash -p nc4nix
+#! nix-shell -I nixpkgs=../../../.. -i bash -p nc4nix jq
```
Comment (dotlambda) [inline pkgs/servers/nextcloud/packages/generate.sh:2]: This should be a separate commit.

**Review by dotlambda — COMMENTED** [review]
I'm sorry this wasn't merged sooner.
Please update to include the package for Nextcloud 28.

File: pkgs/servers/nextcloud/packages/generate.sh (hunk 2 above)
Comment (dotlambda) [reply pkgs/servers/nextcloud/packages/generate.sh]: Please don't drop this change. It should be made but just in a separate commit.

Comment (flandweber) [reply pkgs/servers/nextcloud/packages/generate.sh]: It used to be, together with yet another commit that updated all packages, but rebasing became too annoying which is why I squashed it.
